### PR TITLE
[PLANNER-1040] optashift-employee-rostering-shared persistence.xml parameterizing

### PR DESCRIPTION
**Pull request by calvinzhuca**
Signed-off-by: calvin zhu <calvinzhu@hotmail.com>

### CODE REVIEW

**Comment by calvinzhuca** [comment]
For https://issues.jboss.org/browse/PLANNER-1040
Update the persistence.xml, adding parameters to allow in openshift it can use external db like Postgresql instead of just default H2

**Comment by ge0ffrey** [comment]
ok to test

**Comment by ge0ffrey** [comment]
jenkins retest this and stop giving false negatives

File: optashift-employee-rostering-shared/src/main/resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -3,15 +3,17 @@
              version="2.1" xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/persistence http://xmlns.jcp.org/xml/ns/persistence/persistence_2_1.xsd">
   <persistence-unit name="optashift-employee-rostering-persistence-unit" transaction-type="JTA">
     <description>Hibernate Persistence Unit</description>
-    <jta-data-source>java:jboss/datasources/ExampleDS</jta-data-source>
+    <jta-data-source>${optashift.rostering.persistence.datasource:java:jboss/datasources/ExampleDS}</jta-data-source>
```
Comment (Christopher-Chianelli) [inline optashift-employee-rostering-shared/src/main/resources/META-INF/persistence.xml]: Causing a build failure for me; need to set the default value; should be ${optashift.rostering.persistence.datasource} and you set <optashift.rostering.persistence.datasource>java:jboss/datasources/ExampleDS</optashift.rostering.persistence.datasource> to properties

File: optashift-employee-rostering-shared/src/main/resources/META-INF/persistence.xml [hunk 2]
```diff
@@ -3,15 +3,17 @@
...
     <exclude-unlisted-classes>false</exclude-unlisted-classes>
     <properties>
       <property name="hibernate.hbm2ddl.auto" value="create"/>
-      <!-- <property name="hibernate.show_sql" value="true"/> -->
+      <property name="hibernate.show_sql" value="false"/> 
       <property name="hibernate.format_sql" value="true"/>
       <property name="hibernate.transaction.flush_before_completion" value="true"/>
-      <property name="hibernate.dialect" value="org.hibernate.dialect.H2Dialect"/>
+      <property name="hibernate.dialect" value="${optashift.rostering.persistence.dialect:org.hibernate.dialect.H2Dialect}"/>
```
Comment (Christopher-Chianelli) [inline optashift-employee-rostering-shared/src/main/resources/META-INF/persistence.xml]: See above (change to ${optashift.rostering.persistence.dialect} and set the default value in properties

File: optashift-employee-rostering-shared/src/main/resources/META-INF/persistence.xml (hunk 1 above)
Comment (Christopher-Chianelli) [reply optashift-employee-rostering-shared/src/main/resources/META-INF/persistence.xml]: Well, we need some way to tell wildfly in "wildfly:start wildfly:deploy" to inject those variables (when attempted to run, wildfly reports: "failed to launch service {...${optashift.rostering.persistence.datasource:java:jboss/datasources/ExampleDS}")

Comment (Christopher-Chianelli) [reply optashift-employee-rostering-shared/src/main/resources/META-INF/persistence.xml]: (which should have expanded to "java:jboss/datasources/ExampleDS" (in other words, the wildfly plugin does not expand/replace/do whatever to the variable)

Comment (calvinzhuca) [reply optashift-employee-rostering-shared/src/main/resources/META-INF/persistence.xml]: I tested on EAP 7, used this setting in standalone-openshift.xml, it should work in eap6 and wildfly 10 as well. change this from false to true, it allows parameter replacement.
<spec-descriptor-property-replacement>false</spec-descriptor-property-replacement>

Comment (calvinzhuca) [reply optashift-employee-rostering-shared/src/main/resources/META-INF/persistence.xml]: BTW, this is where I set the spec-descriptor-property-replacement value to true 

https://github.com/calvinzhuca/jboss-kie-modules/blob/master/os.optashift/added/openshift-launch.sh

**Comment by Christopher-Chianelli** [comment]
@ge0ffrey @calvinzhuca Since you cannot do PR's on PR AFAIK, going to open another PR based on your PR where I changed the standalone.xml used by the wildfly plugin.

**Comment by ge0ffrey** [comment]
This PR has been replaced by this PR https://github.com/kiegroup/optaweb-employee-rostering/pull/178 
Use "mvn -DproductizedOpenshift ..." to get a war in which the persistence.xml uses

<jta-```
data-source>${org.optaweb.employeerostering.persistence.datasource:java:jboss/datasources/ExampleDS}</jta-data-source>
<property name="hibernate.dialect" value="${org.optaweb.employeerostering.persistence.dialect:org.hibernate.dialect.H2Dialect}"/>
```

**Pull request closed without merging**